### ISSUE TITLE
[9.5.3] Important Bug : Loop on KnowbaseItem.class.php

### ISSUE THREAD

**Issue by tsmr** [glpi-project/glpi#8385]
Error SQL :

```
[2020-12-04 15:49:10] glpisqllog.ERROR: DBmysql::query() in /usr/share/glpi/inc/dbmysql.class.php line 306
  *** MySQL query error:
  SQL: SELECT `glpi_knowbaseitems`.*, `glpi_knowbaseitemcategories`.`completename` AS `category`, COUNT(`glpi_knowbaseitems_users`.`id`) + COUNT(`glpi_groups_knowbaseitems`.`id`) + COUNT(`glpi_knowbaseitems_profiles`.`id`) + COUNT(`glpi_entities_knowbaseitems`.`id`) AS `visibility_count`, (MATCH(`glpi_knowbaseitems`.`name`, `glpi_knowbaseitems`.`answer`)
                           AGAINST('Liste* Filiales*  *' IN BOOLEAN MODE) ) AS SCORE  FROM `glpi_knowbaseitems` LEFT JOIN `glpi_knowbaseitems_users` ON (`glpi_knowbaseitems_users`.`knowbaseitems_id` = `glpi_knowbaseitems`.`id`) LEFT JOIN `glpi_groups_knowbaseitems` ON (`glpi_groups_knowbaseitems`.`knowbaseitems_id` = `glpi_knowbaseitems`.`id`) LEFT JOIN `glpi_knowbaseitems_profiles` ON (`glpi_knowbaseitems_profiles`.`knowbaseitems_id` = `glpi_knowbaseitems`.`id`) LEFT JOIN `glpi_entities_knowbaseitems` ON (`glpi_entities_knowbaseitems`.`knowbaseitems_id` = `glpi_knowbaseitems`.`id`) LEFT JOIN `glpi_knowbaseitemcategories` ON (`glpi_knowbaseitemcategories`.`id` = `glpi_knowbaseitems`.`knowbaseitemcategories_id`) WHERE (((`glpi_knowbaseitems`.`name` LIKE '%Liste Filiales%') OR (`glpi_knowbaseitems`.`answer` LIKE '%Liste Filiales%'))) AND ((((`glpi_knowbaseitems`.`begin_date` IS NULL) OR (`glpi_knowbaseitems`.`begin_date` < NOW()))) AND (((`glpi_knowbaseitems`.`end_date` IS NULL) OR (`glpi_knowbaseitems`.`end_date` > NOW())))) GROUP BY `glpi_knowbaseitems`.`id`, `glpi_knowbaseitemcategories`.`completename` ORDER BY `SCORE` DESC
  Error: syntax error, unexpected $end, expecting FTS_TERM or FTS_NUMB or '*'
  Backtrace :
  inc/dbmysqliterator.class.php:95                   
  inc/dbmysql.class.php:858                          DBmysqlIterator->execute()
  inc/knowbaseitem.class.php:1437                    DBmysql->request()
  inc/knowbase.class.php:133                         KnowbaseItem::showList()
  inc/knowbase.class.php:82                          Knowbase::showSearchView()
  inc/commonglpi.class.php:637                       Knowbase::displayTabContentForItem()
  ajax/common.tabs.php:92                            CommonGLPI::displayStandardTab()
```

![loop](https://user-images.githubusercontent.com/13178158/101180036-5a474780-364b-11eb-9e4c-c05a8318e119.png)

I have commented this for the moment

```
if ($output_type == Search::HTML_OUTPUT) {
               echo "<td class='center'>";
               $j=0;
               $iterator = $DBread->request([
                  'FIELDS' => 'documents_id',
                  'FROM'   => 'glpi_documents_items',
                  'WHERE'  => [
                     'items_id'  => $data["id"],
                     'itemtype'  => 'KnowbaseItem'
                  ]
               ] + getEntitiesRestrictCriteria());
               while ($docs = $iterator->next()) {
                  $doc = new Document();
                  $doc->getFromDB($docs["documents_id"]);
                  echo $doc->getDownloadLink();
                  $j++;
                  if ($j > 1) {
                     echo "<br>";
                  }
               }
               echo "</td>";
            }
```

**Comment by marcelsilvaa** [comment]
Good Morning,

What was the solution you used to work around the bug?

I have the same problem.

**Comment by bacus99** [comment]
Removing this change correct the issue
![image](https://user-images.githubusercontent.com/11425241/101489366-da60fc00-392e-11eb-91f9-fe19bb0c9905.png)

**Comment by marcelsilvaa** [comment]
Solved, thank you!

**Comment by Electroni-K** [comment]
Thanks, it works with your tips @bacus99

**Comment by andrepetinga** [comment]
Same problem here. Removing the change seems to work.
Thanks

**Comment by cedric-anne** [comment]
Hi,

Mentionned SQL error and error on document list are not related.

The best way to fix document list is #8431 . Removing entity restrictions to fix a bug is never a good idea :/.

Regards
